src/core/shared_memory_manager.cc

```diff
@@ -98,12 +103,62 @@ UnmapSharedMemory(void* mapped_addr, size_t byte_size)
   return Status::Success;
 }
 
+#ifdef TRTIS_ENABLE_GPU
+Status
+OpenCUDAIPCRegion(cudaIpcHandle* shm_cuda_rep, void** data_ptr)
+{
+  cudaEvent_t event;
+  int previous_device;
+  cudaGetDevice(&previous_device);
+
+  cudaSetDevice(shm_cuda_rep->device);
+
+  // get cuda event and synchronize
+  cudaError_t err = cudaIpcOpenEventHandle(&event, shm_cuda_rep->eventHandle);
+  if (err != cudaSuccess) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "failed to get event handle: " + std::string(cudaGetErrorString(err)));
+  }
+
+  err = cudaEventSynchronize(event);
+  if (err != cudaSuccess) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "failed to synchronize event: " + std::string(cudaGetErrorString(err)));
+  }
+  // allocate data on the gpu and read IPC data into it
+  err = cudaMalloc(data_ptr, shm_cuda_rep->byte_size);
```

Comment:
This part and the PR comment below together don't make sense to me. Why the server is responsible for allocating GPU memory and create a IPC handle from it? Shouldn't this be done on the client-side and the server just opens the handle accordingly?

Reply:
You're right, this malloc is unnecessary, the client allocated address for the IPC handle will be used instead.